src/plugins/discover/public/components/discover_container/discover_container.tsx

```diff
@@ -59,23 +59,17 @@ export const DiscoverContainerInternal = ({
   stateStorageContainer,
   isLoading = false,
 }: DiscoverContainerInternalProps) => {
-  const [discoverServices, setDiscoverServices] = useState<DiscoverServices>();
+  const [discoverServices] = useState<DiscoverServices>(() => getDiscoverServices());
```

Comment:
**nit**: as there are no changes on this state and is only used in computing the final services, you could remove this state and directly get the services in the useMemo:
```tsx
const services = useMemo<DiscoverServices>(() => {
  return {
    ...getDiscoverServices(),
    ...overrideServices,
    getScopedHistory: <T>() => scopedHistory as ScopedHistory<T | undefined>,
  };
}, [getDiscoverServices, overrideServices, scopedHistory]);
```

Reply:
Agreed, this makes more sense. Updated here: https://github.com/elastic/kibana/pull/181259/commits/2c013157f8bb67c5ade1444991c1d0f8c6579999.